docroot/modules/custom/sitenow_migrate/src/Plugin/migrate/CreateMediaTrait.php

```diff
@@ -83,11 +83,13 @@ public function createMediaEntity($fid, array $meta = [], $owner_id = 1) {
         case 'application':
         case 'document':
         case 'file':
+          // Set the description as either the title or filename.
+          $description = (isset($meta['title'])) ? $meta['title'] : $meta['filename'];
```

Comment:
Can the description just be blank if a title doesn't exist? This is not a required field and media's own logic will display the filename unless a description is provided.

```
if (isset($meta['title'])) {
  $media_entity['field_media_file']['description'] = $description;
}
```

Reply:
Hrm...In most cases you're right, and it should be handled like that, but for IISC in particular, the human readable name actually comes in as the filename. That worked out alright when it was just setting a name that was needed anyway, but this is a little different...

Reply:
I updated this in...a way? It feels very hackey, but it "works." If we have an actual title, we set it to the description for use in displays (in addition to the name for use in the media browser), and if we don't then we check if the filename gathered in the meta is different from the actual filename, to use that instead. And if it is the same, then we don't bother setting the description.

```
          // If we have a title,
          // go ahead and set it as the description
          // so it can be used in displays.
          if (isset($meta['title'])) {
            $media_entity['field_media_file']['description'] = $meta['title'];
          }
          else {
            // If we didn't have a title, check if we had
            // a human readable filename to use for the description
            // (that doesn't match the true filename, which
            // would be used in displays with an empty description.
            $uri_filename = explode('/', $meta['file_uri']);
            $uri_filename = end($uri_filename);
            if ($meta['filename'] != $uri_filename) {
              $media_entity['field_media_file']['description'] = $meta['filename'];
            }
```

Reply:
Hrm. You lost me, but I trust you. Can you update the test instructions to surface one of these instances?